clients/java/src/main/java/io/odpf/raccoon/client/rest/RestClient.java

```diff
@@ -0,0 +1,145 @@
+package io.odpf.raccoon.client.rest;
+
+import com.google.api.client.http.HttpStatusCodes;
+import com.google.common.io.ByteStreams;
+import com.google.protobuf.ByteString;
+import io.odpf.proton.raccoon.Event.Builder;
+import io.odpf.proton.raccoon.SendEventRequest;
+import io.odpf.proton.raccoon.SendEventResponse;
+import io.odpf.raccoon.client.RaccoonClient;
+import io.odpf.raccoon.model.Event;
+import io.odpf.raccoon.model.Response;
+import io.odpf.raccoon.model.Result;
+import io.odpf.raccoon.wire.JsonWire;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.ServiceUnavailableRetryStrategy;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ByteArrayEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.protocol.HttpContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.UUID;
+
+/**
+ * Class for the http client implementation.
+ */
+public class RestClient implements RaccoonClient {
```

Comment:
This RestClient could have default access only and moved to the client package. Reason: Since a CLientFactory is already provided, a public access on this class would make this class instantiation redundant with the factory.

IMO: Only the factory & restconfig class could be public and rest impls., can be default access

Reply:
Got it, will hide the Rest Implementation, thanks!